coap-core/src/main/java/com/mbed/coap/server/Cache.java

```diff
@@ -0,0 +1,22 @@
+/**
+ * Copyright (C) 2011-2018 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mbed.coap.server;
+
+public interface Cache<K, V> {
```

Comment:
As already discussed, this needs better name. SimpleConcurrentMap? SelfExpiringMap? SelfInvalidatingMap?

Reply:
And maybe add some documentation/notes that this interface represents a map that has implementation specific way of removing/expiring/invalidating entries.

DuplicateDetectableMap could be also one name.

Reply:
PutOnlyMap